README.md

```diff
@@ -33,69 +33,69 @@ The following code example demonstrates how to list paired Nuimos, connect a Nui
-        var options = 2;
-        _nuimoController?.DisplayLedMatrixAsync(new NuimoLedMatrix(matrixString), displayInterval, options);
-    }
-
-    private void OnBatteryPercentage(int batteryPercentage)
-    {
-        Debug.WriteLine("Battery percentage: " + batteryPercentage);
-    }
-
-    private async void OnLedMatrixDisplayed()
+    public class Demo
     {
-        Debug.WriteLine("LED matrix displayed");
+        private readonly PairedNuimoManager _pairedNuimoManager = new PairedNuimoManager();
+        private INuimoController _nuimoController;
+
+        private async void GetPairedNuimos()
+        {
+            var nuimoControllers = await _pairedNuimoManager.ListPairedNuimosAsync();
+            _nuimoController = nuimoControllers.ElementAt(0);
+        }
+
+        private async void Connect()
+        {
+            var isConnected = await _nuimoController.ConnectAsync();
+        }
+
+        private void AddDelegates()
+        {
+            _nuimoController.GestureEventOccurred += OnNuimoGestureEvent;
+            _nuimoController.FirmwareVersionRead += OnFirmwareVersion;
+            _nuimoController.ConnectionStateChanged += OnConnectionState;
+            _nuimoController.BatteryPercentageChanged += OnBatteryPercentage;
+            _nuimoController.LedMatrixDisplayed += OnLedMatrixDisplayed;
+        }
+
+        private void OnNuimoGestureEvent(INuimoController controller, NuimoGestureEvent nuimoGestureEvent)
+        {
+            Debug.WriteLine("Event: " + nuimoGestureEvent.Gesture + ", " + nuimoGestureEvent.Value);
+        }
+
+        private void OnFirmwareVersion(INuimoController controller, string firmwareVersion)
+        {
+            Debug.WriteLine(firmwareVersion);
+        }
+
+        private void OnConnectionState(INuimoController controller, NuimoConnectionState nuimoConnectionState)
+        {
+            Debug.WriteLine("Connection state: " + nuimoConnectionState);
+
+            var displayInterval = 5.0;
+            var matrixString = "         " +
+                                    "         " +
+                                    " ..   .. " +
+                                    "   . .   " +
+                                    "    .    " +
+                                    "   . .   " +
+                                    " ..   .. " +
+                                    "         " +
+                                    "         ";
+            var options = 2;
+            _nuimoController?.DisplayLedMatrixAsync(new NuimoLedMatrix(matrixString), displayInterval, options);
+        }
+
+        private void OnBatteryPercentage(INuimoController controller, int batteryPercentage)
+        {
+            Debug.WriteLine("Battery percentage: " + batteryPercentage);
+        }
+
+        private async void OnLedMatrixDisplayed(INuimoController controller)
+        {
+            Debug.WriteLine("LED matrix displayed");
+        }
     }
-}
```

Comment:
Oh, oops. I had it through my IDE to make sure it was completely valid code. I'll fix it when I get to my pc again tomorrow afternoon